controllers/medusa/medusatask_controller.go

```diff
@@ -0,0 +1,470 @@
+				succeeded := false
+				err := prepareRestore(ctx, task, &pod, r.ClientFactory)
+				if err == nil {
+					logger.Info("finished prepare restore", "CassandraPod", pod.Name)
+					succeeded = true
+				} else {
+					logger.Error(err, "prepare restore failed", "CassandraPod", pod.Name)
+				}
+				backupMutex.Lock()
+				defer backupMutex.Unlock()
+				defer wg.Done()
+				task.Status.InProgress = utils.RemoveValue(task.Status.InProgress, pod.Name)
+				if succeeded {
+					operationResponse := medusav1alpha1.TaskResult{
+						PodName: pod.Name,
+					}
+					task.Status.Finished = append(task.Status.Finished, operationResponse)
+				} else {
+					task.Status.Failed = append(task.Status.Failed, pod.Name)
+				}
+			}()
+		}
+		wg.Wait()
+		logger.Info("finished task operations")
+		if err := r.Status().Patch(context.Background(), task, patch); err != nil {
+			logger.Error(err, "failed to patch status", "MedusaTask", fmt.Sprintf("%s/%s", task.Spec.Operation, task.Namespace))
+		}
+	}()
+
+	return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+}
+
+func (r *MedusaTaskReconciler) syncOperation(ctx context.Context, task *medusav1alpha1.MedusaTask, pods []corev1.Pod, logger logr.Logger) (reconcile.Result, error) {
+	logger.Info("Starting sync operation")
+	syncStartTime := metav1.Now()
+	for _, pod := range pods {
+		logger.Info("Listing Backups", "CassandraPod", pod.Name)
+		if remoteBackups, err := getBackups(ctx, &pod, r.ClientFactory); err != nil {
+			logger.Error(err, "failed to list backups", "CassandraPod", pod.Name)
+		} else {
+			for _, backup := range remoteBackups {
+				logger.Info("Syncing Backup", "Backup", backup.BackupName)
+				// Create backups that should exist but are missing
+				backupKey := types.NamespacedName{Namespace: task.Namespace, Name: backup.BackupName}
+				backupResource := &medusav1alpha1.CassandraBackup{}
+				if err := r.Get(ctx, backupKey, backupResource); err != nil {
+					if errors.IsNotFound(err) {
+						// Backup doesn't exist, create it
+						logger.Info("Creating Cassandra Backup", "Backup", backup.BackupName)
+						startTime := metav1.Unix(backup.StartTime, 0)
+						finishTime := metav1.Unix(backup.StartTime+1000, 0)
+						backupResource = &medusav1alpha1.CassandraBackup{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      backup.BackupName,
+								Namespace: task.Namespace,
+							},
+							Spec: medusav1alpha1.CassandraBackupSpec{
+								Name:                backup.BackupName,
+								CassandraDatacenter: task.Spec.CassandraDatacenter,
+								Type:                medusav1alpha1.BackupType("differential"),
+								CreatedBySync:       true,
+							},
+						}
+						if err := r.Create(ctx, backupResource); err != nil {
+							logger.Error(err, "failed to create backup", "CassandraBackup", backup.BackupName)
+							return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
+						} else {
+							logger.Info("Created Cassandra Backup", "Backup", backupResource)
```

Comment:
I know you talked about why you have the for loop on our call, but I don't recall the details so bear with my question if it was already asked. I don't think we need to fetch the CassandraBackup object at all. The `Create` call above at line 307 will initialize the necessary metadata fields to apply a follow up patch. In particular `ResourceVersion` will be set. You should be able to simply follow up the `Create` with the `Status().Patch()` call.

Reply:
I'll try again, but I remember Status wouldn't be updated when I tried to update it without reading the object back from storage. This is why I went with this solution (and the for loop was there to retry because of cache issues that wouldn't return the object from the first attempt).
I'll post back once I have some results.

Reply:
You were right @jsanda, I can patch the status straight away without needing to re-read the object from the server.
The code is much simpler now 👍